src/org/labkey/test/tests/SampleTypeNameExpressionTest.java

```diff
@@ -130,6 +135,54 @@ public void testParentAliasExpression()
                 "Parent", "Jessi");
     }
 
+    // Issue 42857: samples: bulk import with name expression containing lookup fails to convert lookup string value
+    @Test
+    public void testLookupNameExpression() throws Exception
+    {
+        String lookupList = "Colors";
+        FieldDefinition.LookupInfo colorsLookup = new FieldDefinition.LookupInfo(getProjectName(), "lists", lookupList);
+        String nameExpSamples = "NameExpressionSamples";
+
+        // begin by creating a lookupList of colors, the sampleType will reference it
+        TestDataGenerator colorsGen = new TestDataGenerator(colorsLookup)
+                .withColumns(List.of(new FieldDefinition("ColorName", FieldDefinition.ColumnType.String),
+                        new FieldDefinition("ColorCode", FieldDefinition.ColumnType.String)));
+        colorsGen.addCustomRow(Map.of("ColorName", "green", "ColorCode", "gr"));
+        colorsGen.addCustomRow(Map.of("ColorName", "yellow", "ColorCode", "yl"));
+        colorsGen.addCustomRow(Map.of("ColorName", "red", "ColorCode", "rd"));
+        colorsGen.addCustomRow(Map.of("ColorName", "blue", "ColorCode", "bl"));
+        colorsGen.createList(createDefaultConnection(), "Key");
+        colorsGen.insertRows();
+
+        String pasteData = "ColorLookup\tNoun\n" +
+                "red\tryder\n" +
+                "green\tgiant\n" +
+                "blue\tangel\n" +
+                "yellow\tjersey";
+
+        // now create a sampleType with a Color column that looks up to Colors
+        var sampleTypeDef = new SampleTypeDefinition(nameExpSamples)
+                .setFields(List.of(new FieldDefinition("ColorLookup", colorsLookup).setRangeURI(FieldDefinition.ColumnType.Integer.getRangeURI()),
+                        new FieldDefinition("Noun", FieldDefinition.ColumnType.String)))
```

Comment:
The correct pattern to set the type of a lookup is to call `setTableType`. This `setRangeURI` isn't necessary.